coral/src/app/features/topics/acl-request/forms/TopicProducerForm.tsx

```diff
@@ -128,11 +130,13 @@ const TopicProducerForm = ({
           </GridItem>
 
           <GridItem colSpan={"span-2"}>
-            <TextInput
-              name="transactionalId"
-              labelText="Transactional ID"
-              placeholder="Necessary for exactly-once semantics on producer"
-            />
+            {isAivenCluster ? null : (
```

Comment:
`{!isAivenCluster && <TextInput ...}` would be more idiomatic conditional rendering.

Reply:
Eyyy it's the time for pedantry!

I usually prefer the approach currently in the code for two reasons:
- `&&` means that we return `false`, which does work in React as will render nothing, but is imo less correct than explicitly returning `null`
- `!isAivenCluster` is negative logic, which I think is usually harder to read (fractionally)

I literally do not care though, so I will implement the way you suggest, which does have the merit of looking neat ^^

Reply:
Official documentation describes the `&&` approach [[ref](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)] :)